openapi2beans/pkg/generator/schema2package.go

```diff
@@ -0,0 +1,200 @@
+}
+
+func possibleValuesToEnumValues(possibleValues map[string]string) (enumValues []string) {
+	for _, value := range possibleValues {
+		enumValues = append(enumValues, value)
+	}
+	return enumValues
+}
+
+func retrieveDataMembersFromSchemaType(schemaType *SchemaType) (dataMembers []*DataMember, requiredMembers []*RequiredMember, constantDataMembers []*DataMember){
+	for _, property := range schemaType.properties {
+		var constVal string
+		name := property.name
+		description := strings.Split(property.description, "\n")
+		if len(description) == 1 && description[0] == "" {
+			description = nil
+		} else if len(description) > 1 {
+			description = description[:len(description)-2]
+		}
+		if property.IsConstant() {
+			posVal := possibleValuesToEnumValues(property.GetPossibleValues())
+			name = convertToConstName(name)
+			constVal = convertConstValueToJavaReadable(posVal[0], property.typeName)
+
+			constDataMember := DataMember {
+				Name: name,
+				CamelCaseName: convertToCamelCase(name),
+				MemberType: propertyToJavaType(property),
+				Description: description,
+				ConstantVal: constVal,
+			}
+
+			constantDataMembers = append(constantDataMembers, &constDataMember)
+
+		} else {
+
+			dataMember := DataMember {
+				Name: name,
+				CamelCaseName: convertToCamelCase(name),
+				MemberType: propertyToJavaType(property),
+				Description: description,
+				ConstantVal: constVal,
+			}
+			dataMembers = append(dataMembers, &dataMember)
+				
+			if property.IsSetInConstructor() {
+				requiredMember := RequiredMember {
+					DataMember: &dataMember,
+				}
+				requiredMembers = append(requiredMembers, &requiredMember)
+			}
+		}
+		
+	}
+	sort.SliceStable(dataMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(dataMembers[i], dataMembers[j])})
+	sort.SliceStable(requiredMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(requiredMembers[i].DataMember, requiredMembers[j].DataMember)})
+	sort.SliceStable(constantDataMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(constantDataMembers[i], constantDataMembers[j])})
+	if requiredMembers != nil {
+		requiredMembers[0].IsFirst = true
+	}
+	return dataMembers, requiredMembers, constantDataMembers
+}
+
+func propertyToJavaType(property *Property) string {
+	javaType := ""
+	if property.IsReferencing() || property.typeName == "object" || property.IsEnum() {
+		javaType = property.resolvedType.name
+	} else {
+		if property.typeName == "string" {
+			javaType = "String"
+		} else if property.typeName == "integer" {
+			javaType = "int"
+		} else if property.typeName == "number" {
+			javaType = "double"
+		}else if property.typeName == "" {
+			javaType = "Object"
+		} else {
+			javaType = property.typeName
+		}
+	}
+
+	if property.IsCollection() {
+		dimensions := property.cardinality.max / 128
+		for range dimensions {
+			javaType += "[]"
+		}
+	}
+
+	return javaType
+}
+
+func convertToCamelCase(name string) string {
```

Comment:
How can this be desireable ?
I could believe a camelcase to lower or camelcase to uppercase, but having an initial letter lower-case and all the rest uppercase isn't really camel case ?
I guess I'm missing the context. A comment might help perhaps.

Reply:
ah, should be Pascal case, I got them mixed up my bad, I'll add comments and change the name, it is generally for use in getters and setters and enum java file names. i.e. templates would have (originally) GetaIntVar or enum myEnum {}.

Reply:
the templates would now have getaIntVar (getters and setters begin with lowercase as of this comment and previous comment)